### PR TITLE
#5024 Cannot convert duration type when shabondi image startup

### PR DESCRIPTION
**Pull request by vitojeng**
### Checklist
- [x] [Review Contribution Guidelines](https://ohara.readthedocs.io/en/latest/contributing.html)
- [x] Add Reviewers (see [Authors](https://github.com/oharastream/ohara#ohara-team))
- [x] Add Assignees (of course, you are the assignee by default!)
- [x] DON'T set a Milestone
- [x] DON'T add any labels (PR labels will be added automatically by Jenkins)
- [x] Link to an issue(You can do so by using the required by keyword: required by #5024)
- [ ] Add unit tests (or please explain why this PR is not worth having new tests)
- [ ] Pass QA (Sometimes you may encounter an existing flaky test which would obstruct you from getting QA passed. If the failed tests are unrelated, you can add a comment in the PR thread)

### CODE REVIEW

File: ohara-agent/src/main/scala/oharastream/ohara/agent/ShabondiCollie.scala [hunk 1]
```diff
@@ -87,7 +87,7 @@ trait ShabondiCollie extends Collie {
                 val arguments = creation.settings.map {
                   case (k, v) =>
                     val value = v match {
-                      case JsString(s) => s
+                      case JsString(s) => ShabondiUtils.escape(s)
```
Comment (chia7712) [inline ohara-agent/src/main/scala/oharastream/ohara/agent/ShabondiCollie.scala:90]: stream也需要類似的處理

File: ohara-shabondi/src/main/scala/oharastream/ohara/shabondi/common/ShabondiUtils.scala [hunk 2]
```diff
@@ -23,16 +23,22 @@ import scala.jdk.CollectionConverters._
 object ShabondiUtils {
   private val log = Logger(ShabondiUtils.getClass)
 
-  private val ESCAPE_STRING = "_____"
+  private val ESCAPE_STRING1 = "_____"
+  private val ESCAPE_STRING2 = "~~~~~"
 
   def escape(value: String): String = {
```
Comment (chia7712) [inline ohara-shabondi/src/main/scala/oharastream/ohara/shabondi/common/ShabondiUtils.scala:29]: 不是要採取base64的編碼嗎?

Comment (vitojeng) [reply ohara-shabondi/src/main/scala/oharastream/ohara/shabondi/common/ShabondiUtils.scala:29]: base64 後面可能會出現 `=`, 發現在 CommonUtils.parse() 可能會有問題
因為不想改太多, 就不採用了

Comment (chia7712) [reply ohara-shabondi/src/main/scala/oharastream/ohara/shabondi/common/ShabondiUtils.scala:29]: 如果只是要處理duration的型別，另一種做法是改傳java duration的字串格式，然後用CommonUtils.toDuration去解，```CommonUtils.toDuration```可以處理java and scala的格式

然後我們在0.11.0在來做完整個解法

Comment (vitojeng) [reply ohara-shabondi/src/main/scala/oharastream/ohara/shabondi/common/ShabondiUtils.scala:29]: 嗯... 我想這裡不只針對 duration, 最好是所有參數裡的空白.

Comment (chia7712) [reply ohara-shabondi/src/main/scala/oharastream/ohara/shabondi/common/ShabondiUtils.scala:29]: ok

File: ohara-agent/src/main/scala/oharastream/ohara/agent/ShabondiCollie.scala (hunk 1 above)
Comment (vitojeng) [reply ohara-agent/src/main/scala/oharastream/ohara/agent/ShabondiCollie.scala:90]: done

**Comment by vitojeng** [comment]
retry

**Comment by vitojeng** [comment]
not related error

**Pull request merged**